src/EquationSystem.C

```diff
@@ -294,6 +299,32 @@ EquationSystem::assemble_and_solve(
   update_iteration_statistics(
     linsys_->linearSolveIterations());
   
+  // update equation system nonlinear iteration count
+  if (linsys_->linearSolveIterations() != 0)
```

Comment:
I don't understand why we need an `if` check here. Shouldn't this counter be always updated if we have called the `linsys_->solve` method ?